doc/advanced.rst

```diff
@@ -57,12 +57,15 @@ into the HTML version of your documentation have a particular
 namespace, to avoid collisions with your own files and images.
 
 Our namespace convention is to prefix everything with ``sphx_glr`` and
-change path separators with underscores. This is valid for
-cross-references labels, and images.
+change path separators with underscores. 
 
-So for example if we want to reference the example
-:ref:`sphx_glr_auto_examples_plot_0_sin.py`, we just call
-its reference
+Referencing Examples
+--------------------
+
+The namespace convention mentioned in the paragraph above is valid for
+cross-references labels, and images. So for example if we want to 
+reference the example :ref:`sphx_glr_auto_examples_plot_0_sin.py`, 
+we just call its reference 
```

Comment:
I think I would put the cross referencing part a bit separate because the cross referencing of examples is enabled because we put a reference label at the start of each generated `.rst` file, which is a slightly different topic than how we name generated files.

Maybe:

"Our namespace convention for generated files is to prefix everything with ``sphx_glr`` and end with the name of the file. For example the first image generated by the example ``plot_0_sin.py`` is ``sphx_glr_plot_0_sin_001.png`` and its thumbnail is ``sphx_glr_plot_0_sin_thumb.png``.

I would have the literal include paragraphs, starting with "You can also include part of", follow.

At the end of the section (line 97 in the new commit), before the line ".. _warning_errors:" I would add your new referencing header.

I would call it "cross referencing", to make it consistent with https://sphinx-gallery.github.io/stable/auto_examples/plot_0_sin.html#cross-referencing

```rst
Cross referencing
-----------------

You can also cross reference an example using similar naming convention. For example
if we want to reference the example :ref:`sphx_glr_auto_examples_plot_0_sin.py`,
we just call its reference ``:ref:`sphx_glr_auto_examples_plot_0_sin.py```.
Note that we have included the path to the example file (relative to
the ``conf.py`` file) after ``sphx_glr``. Path separators are replaced with
underscores.
```

Reply:
Hey, I'm not sure if I understood your suggestion correctly, but I added the suggested paragraph and reverted the other changes. I think having a separate paragraph makes total sense, but I'm not sure whether you suggested that I should edit the section before.